runtime/pkg/globutil/globutil.go

```diff
@@ -0,0 +1,28 @@
+package globutil
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// ParseURL splits urls with globs into scheme, hostname and rest of the url(as glob)
+// url.Parse removes `?` considering it is query param
+// `?` is valid meta in glob patterns
+func ParseURL(path string) (string, string, string, error) {
+	u, err := url.Parse(path)
+	if err != nil {
+		return "", "", "", err
+	}
+	// everything after `Host/` is glob
+	_, glob, found := strings.Cut(path, fmt.Sprintf("%s/", u.Host))
+	if !found {
+		return "", "", "", fmt.Errorf("failed to parse url %s", path)
+	}
+
+	if u.Scheme == "" || u.Host == "" {
+		return "", "", "", fmt.Errorf("invalid url %s", path)
+	}
```

Comment:
URLs can be quite weird, I'm a little worried if we might miss some edge cases here. Maybe keep it safe and simple and do something like?

```go
func ParseURL(path string) (string, string, string, error) {
	scheme, path, ok := strings.Cut(path, "://")
	if !ok {
		return nil, fmt.Errorf("failed to parse URL '%q'", path)
	}

	host, path, ok := strings.Cut(path, "/")
	if !ok {
		// This is actually a valid URL, just not a valid object storage URL.
		// Should we return it – or maybe rename function to `ParseBucketURL` or something?
		return nil, fmt.Errorf("failed to parse URL '%q'", path)
	}

	return scheme, host, path
}
```

Reply:
Done. Although I am not sure if this implementation will cover all edge cases (detecting incorrect URLs as incorrect). Anyways in case of any incorrect parsing the cloud storage sdk will return bucket not found error (or similar).